spec/reek/smells/instance_variable_assumption_spec.rb

```diff
@@ -0,0 +1,184 @@
+require_relative '../../spec_helper'
+require_lib 'reek/smells/instance_variable_assumption'
+
+RSpec.describe Reek::Smells::InstanceVariableAssumption do
+  context 'report' do
+    context 'smell line' do
+      it 'should report the core class in the message' do
+        src = <<-EOS
+          class Dummy
+            def test
+              @a = 1
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, lines: [1])
+      end
+    end
+
+    context 'smell context' do
+      it 'should report the class as the context' do
+        src = <<-EOS
+          class Dummy
+            def test
+              @a = 1
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, context: 'Dummy')
+      end
+    end
+
+    context 'smell message' do
+      it 'should report the class and ivars in the message' do
+        message = 'has @a instance variable assumptions'
+
+        src = <<-EOS
+          class Dummy
+            def test
+              @a = 1
+            end
+          end
+        EOS
+
+        expect(src).to reek_of(:InstanceVariableAssumption, message: message)
+      end
+    end
+  end
+
+  context 'counting instance variable assumptions' do
+    it 'should not report for using ivars from initialize' do
+      src = <<-EOS
+        class Dummy
+          def initialize
+            @a = 1
+          end
+
+          def test
+            @a = 1
+          end
+        end
+      EOS
+
+      expect(src).not_to reek_of(:InstanceVariableAssumption)
+    end
+
+    it 'should not report when lazy initializing' do
+      src = <<-EOS
+        class Dummy
+          def test
+            @a ||= 1
+          end
+        end
+      EOS
+
+      expect(src).not_to reek_of(:InstanceVariableAssumption)
+    end
+
+    it 'should report when making instance variable assumption' do
+      src = <<-EOS
+        class Dummy
+          def test
+            @a == 1
+          end
+        end
+      EOS
+
+      expect(src).to reek_of(:InstanceVariableAssumption)
+    end
+
+    it 'should report any initialization outside initialize' do
+      src = <<-EOS
+        class Dummy
+          def initialize
+            @a = 1
+          end
+
+          def test
+            @a = 1
```

Comment:
Why do you even have "@a" here? That's confusing. Can't we just remove it here and the "initialize" completely?